__tests__/app.test.js

```diff
@@ -77,4 +77,100 @@ describe('GET /api/topics', () => {
+            expect(body).toHaveProperty('article');
+        });
+    });
+
+    test('each article should have correct object layout', () => {
+        const objectLayout = {
+            article_id: expect.any(Number),
+            title: expect.any(String),
+            topic: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url: expect.any(String)
+        };
+
+        const supertestRequests = [];
+
+        for (let i=1; i<=13; i++) {  // 13 articles in test db
+            const pendingRequest = request(app)
+            .get(`/api/articles/${i}`)
+            .expect(200);
+            
+            const article = new Promise((resolve, reject) => {
+                pendingRequest.then(({ body }) => {
+                    resolve(body.article);
+                })
+                .catch(err => reject(err));
+            });
+
+            supertestRequests.push(article);
+        }
+
+        return Promise.all(supertestRequests)
+        .then(articles => {
+            for (const article in articles) {
+                expect(articles[article]).toMatchObject(objectLayout);
+            }
+        });
+    });
+
+    test('response data should match the provided seed data', () => {
+        const seedArticles = data.articleData;
+
+        const supertestRequests = [];
+
+        for (let i=0; i<seedArticles.length; i++) {
+            const seedArticle = seedArticles[i];
+
+            // timestamps in SQL have not been stored as UTC, need converting
+
+            const timestamp = new Date(seedArticle.created_at);
+            const localTimestamp = new Date(timestamp.getTime() + timestamp.getTimezoneOffset()*60000)
+            seedArticle.created_at = localTimestamp.toISOString();
+
+            const pendingRequest = request(app)
+            .get(`/api/articles/${i+1}`)
+            .expect(200);
+            
+            const articles = new Promise((resolve, reject) => {
+                pendingRequest.then(({ body }) => {
+                    const article = body.article;
+                    resolve({ seedArticle, article });
+                })
+                .catch(err => reject(err));
+            });
+
+            supertestRequests.push(articles);
+        }
+
+        return Promise.all(supertestRequests)
+        .then((supertestResponses) => {
+            for (const response of supertestResponses) {
+                expect(response.article).toMatchObject(response.seedArticle);
+            }
+        });
+    });
+
+    describe('error handling', () => {
+        it('should return a http 404 error if article not in database', () => {
+            return request(app)
+            .get('/api/articles/99')
+            .expect(404)
+            .then(({ body }) => {
+                expect(body.msg).toBe('article not found');
+            });
+        });
```

Comment:
Great to have a 404 test here, what if I made a request to /api/articles/banana? What error would we get here? Could we add a test for it?